components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SQLQueries.java

```diff
@@ -500,6 +500,59 @@ public class SQLQueries {
             "JOIN IDN_OAUTH2_ACCESS_TOKEN_SCOPE ON ACCESS_TOKEN_TABLE.TOKEN_ID = " +
             "IDN_OAUTH2_ACCESS_TOKEN_SCOPE.TOKEN_ID";
 
+    /**
+     * SQL query to retrieve access tokens for a given consumer key and scope.
+     * It selects various attributes of access tokens from the IDN_OAUTH2_ACCESS_TOKEN table.
+     * The query involves joining multiple tables:
+     * - It selects access tokens from a subquery filtering by consumer key ID and token state.
+     * - It joins the resulting access token table with the IDP table based on the IDP ID.
+     * - It left joins the access token scope table to retrieve token scope information.
+     * The query parameters are placeholders for the consumer key, tenant ID, token state, and token scope.
+     */
+    public static final String  GET_ACCESS_TOKENS_FOR_CONSUMER_KEY_AND_SCOPE =
+            "SELECT " +
+                    "ACCESS_TOKEN, " +
```

Comment:
Why we need to select all these data?

Reply:
Currently this query is used in revocation procedure, but scope of this query is GET ACCESS TOKENS FOR CONSUMER KEY AND SCOPE, this query can be used in future several RBAC operations and token binding reference operations.

I added all these details cause it will used to construct access token DAO which can be used for future references.

Reply:
Let's not fetch all the details if we do not use them in the current code.  If we need it in future, we can simply update the query.